fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/AbstractService.java

```diff
@@ -0,0 +1,274 @@
+     *
+     * @param externalPath the path.
+     */
+    void hasRestrictedPath(final String externalPath) {
+        final String[] pathSegments = externalPath.split("/");
+        if (Arrays.stream(pathSegments).anyMatch(p -> p.startsWith("fedora:"))) {
+            throw new ServerManagedTypeException("Path cannot contain a fedora: prefixed segment.");
+        }
+    }
+
+    /**
+     * Looks through an RdfStream for rdf:types in the LDP namespace or server managed predicates.
+     *
+     * @param model The RDF model.
+     */
+    void checkForSmtsLdpTypes(final Model model) {
+        final StmtIterator it = model.listStatements();
+        while (it.hasNext()) {
+            final Statement st = it.next();
+            if (st.getPredicate().hasURI(rdfType) && st.getObject().isURIResource() &&
+                    st.getObject().toString().startsWith(LDP_NAMESPACE)) {
+                throw new MalformedRdfException("RDF contains a server managed triple or restricted rdf:type");
+            }
+        }
+    }
+
+    /*
+     * This method throws an exception if the arg model contains a triple with 'ldp:hasMemberRelation' as a predicate
+     *   and a server-managed property as the object.
+     *
+     * @param inputModel to be checked
+     * @throws ServerManagedPropertyException on error
+     */
+    protected void ensureValidMemberRelation(final Model inputModel) {
+        // check that ldp:hasMemberRelation value is not server managed predicate.
+        inputModel.listStatements().forEachRemaining((final Statement s) -> {
+            log.debug("statement: s={}, p={}, o={}", s.getSubject(), s.getPredicate(), s.getObject());
+
+            if (s.getPredicate().equals(HAS_MEMBER_RELATION)) {
+                final RDFNode obj = s.getObject();
+                if (obj.isURIResource()) {
+                    final String uri = obj.asResource().getURI();
+
+                    // Throw exception if object is a server-managed property
+                    if (isManagedPredicate.test(createProperty(uri))) {
+                            throw new ServerManagedPropertyException(
+                                    String.format(
+                                            "{0} cannot take a server managed property " +
+                                                    "as an object: property value = {1}.",
+                                            HAS_MEMBER_RELATION, uri));
+                    }
+                }
+            }
+        });
+    }
+
+    /**
+     * This method does two things:
+     * - Throws an exception if an authorization has both accessTo and accessToClass
+     * - Adds a default accessTo target if an authorization has neither accessTo nor accessToClass
+     *
+     * @param fedoraId the fedora Id
+     * @param inputModel to be checked and updated
+     */
+    protected void ensureValidACLAuthorization(final String fedoraId, final Model inputModel) {
```

Comment:
Would it make sense to put this test in the HttpRdfService now as we are parsing the provided RDF there anyways?

Reply:
Hmm, my understanding is that this is very specific to creating/updating WebACs? HttpRdfService is pretty well scoped to translating URIs in RDF bodies at the moment, so I think it's in a good place. To me at least it would make sense to have a service dedicated to updating/validating ACLs, even if it hits the persistence layer like any other RDF resource

Reply:
I've created a follow up ticket to address this https://jira.lyrasis.org/browse/FCREPO-3134